taurus/entity/setters.py

```diff
@@ -13,7 +31,20 @@ def validate_list(obj, typ, trigger=None):
 
 
 def validate_str(obj):
-    """Check that obj is a string and then convert it to unicode."""
+    """
+    Check that obj is a string and then convert it to unicode.
+
+    Parameters
+    ----------
+    obj: Any
+        Object that should be a string
+
+    Returns
+    -------
+    str
+        Return obj if it is a string, otherwise throw ValueError
```

Comment:
Is this the right way to document potential exceptions in this formalism?

Reply:
Good question. Turns out they do have a `Raises` section, which I have now included.